cmd/service-controller/controller.go

```diff
@@ -316,7 +316,7 @@ func (c *Controller) checkServiceFor(desired *ServiceBindings, actual *corev1.Se
 			actual.Spec.Ports[0].TargetPort = intstr.FromInt(desired.ingressPort)
 		}
 	}
-	if !equivalentSelectors(actual.Spec.Selector, kube.GetLabelsForRouter()) {
+	if desired.headless == nil && !equivalentSelectors(actual.Spec.Selector, kube.GetLabelsForRouter()) {
```

Comment:
Strictly speaking I think the test would be:

``` (desired.headless == nil || desired.origin != "") && !equivalentSelectors(...)```

i.e. we could update the service even if it is headless as long as it originated elsewhere. However the fix as is is sufficient for now and fixes a regression so :shipit:!

Reply:
And thanks for fixing this @ajssmith !

Reply:
Iiuc a headless service originating elsewhere would still have the same issue, e.g. we would not want to change the labels from proxy to router.

Reply:
Yes, you are correct. The labels required there would indeed be different.